### PR TITLE
test(extension): first multidelegation full flow test

### PR DESCRIPTION
**Pull request by oldGreg5**
# Checklist

- [ ] JIRA [LW-7654](https://input-output.atlassian.net/browse/LW-7654)
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

First e2e test for multidelegation full flow

[LW-7654]: https://input-output.atlassian.net/browse/LW-7654?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/1374/5795538177/index.html) for [d9032883](https://github.com/input-output-hk/lace/pull/332/commits/d9032883a38ce74f4d27754379ecdedd9bc48d08)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 34     | 2      | 0       | 0     | 36    | ❌     |

### CODE REVIEW

File: packages/staking/src/features/staking/PortfolioBar.tsx [hunk 1]
```diff
@@ -25,8 +25,13 @@ export const PortfolioBar = ({ onStake }: PortfolioBarParams) => {
         <span>{t('portfolioBar.maxPools', { maxPoolsCount: MAX_POOLS_COUNT })}</span>
       </Text.Body.Normal>
       <Flex className={styles.buttons}>
-        <Button.Secondary label="Clear" onClick={portfolioMutators.clearDraft} />
-        <Button.Primary label="Next" icon={<ArrowRight className={styles.nextIcon} />} onClick={onStake} />
+        <Button.Secondary label="Clear" onClick={portfolioMutators.clearDraft} data-testid="portfolioBar-btn-clear" />
```
Comment (wklos-iohk) [inline packages/staking/src/features/staking/PortfolioBar.tsx:28]: 1. Labels should be taken from relevant file with translations.
2. I'd suggest to not use upper-case letters in test ids.

File: packages/staking/src/features/staking/PortfolioBar.tsx [hunk 2]
```diff
@@ -25,8 +25,13 @@ export const PortfolioBar = ({ onStake }: PortfolioBarParams) => {
...
+        <Button.Primary
```
Comment (wklos-iohk) [inline packages/staking/src/features/staking/PortfolioBar.tsx:29]: as above

File: packages/staking/src/features/staking/PortfolioBar.tsx (hunk 1 above)
Comment (oldGreg5) [reply packages/staking/src/features/staking/PortfolioBar.tsx]: That's a small change so I changed to reading from i18n

File: packages/e2e-tests/src/elements/staking/MultidelegationPage.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+  get overviewTab() {
+    return $(this.OVERVIEW_TAB);
+  }
+
+  get browseTab() {
+    return $(this.BROWSE_POOLS_TAB);
+  }
+
+  get delegationcardPoolsValue() {
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/elements/staking/MultidelegationPage.ts:31]: Can we adjust the name to `delegationCardPoolsValue` ?

File: packages/e2e-tests/src/steps/multidelegationSteps.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+Given(/^I click (Overview|Browse pools) tab$/, async (tabToClick: 'Overview' | 'Browse pools') => {
+  await MultidelegationPage.clickOnTab(tabToClick);
+});
+
+Then(/^I wait until delegation info card shows staking to "(\d+)" pool\(s\)$/, async (poolsCount: number) => {
+  await MultidelegationPageAssert.assertSeeStakingOnPoolsCounter(poolsCount);
+});
+
+Then(
+  /^I pick "(\d+)" pools for delegation from browse pools view: "([^"]*)"$/,
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/steps/multidelegationSteps.ts:14]: Why do we have this first parameter ignored? Is it really needed?

File: packages/e2e-tests/src/steps/multidelegationSteps.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  async (_ignored: number, poolsToStake: string) => {
+    await MultidelegationPage.markPoolsForDelegation(poolsToStake);
+  }
+);
+
+Then(
+  /^I click (Next|Clear) button on staking (portfolio bar|manage staking|confirmation)$/,
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/steps/multidelegationSteps.ts:21]: This step is misleading, `clear` button won't be clicked at all according to current implementation

File: packages/e2e-tests/src/elements/staking/MultidelegationPage.ts [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+  get manageStakingBtnNext() {
+    return $(this.MANAGE_STAKING_BTN_NEXT);
+  }
+
+  get confirmationBtnNext() {
+    return $(this.CONFIRMATION_BTN_NEXT);
+  }
+
+  async clickOnTab(tab: string) {
+    await this.overviewTab.waitForClickable();
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/elements/staking/MultidelegationPage.ts:66]: I'd suggest to rewrite it with `switch...case` statement
```
    switch (tab) {
      case 'Overview':
        await this.overviewTab.waitForClickable();
        await this.overviewTab.click();
        break;
      case 'Browse pools':
        await this.browseTab.waitForClickable();
        await this.browseTab.click();
        break;
      default:
        ...
    }
```

File: packages/e2e-tests/src/elements/staking/MultidelegationPage.ts [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+    await this.hoverOverPoolWithName(poolName);
+    await this.markPoolForStaking();
+  }
+
+  private async hoverOverPoolWithName(poolName: string) {
+    const poolItem = await this.getPoolByName(poolName);
+    await poolItem.moveTo();
+  }
+
+  private async markPoolForStaking() {
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/elements/staking/MultidelegationPage.ts:104]: This one seems to be pretty small and used only once in `markStakePoolWithName`, hence they could be merged.

**Review by wklos-iohk — CHANGES_REQUESTED** [review]

File: packages/e2e-tests/src/elements/staking/MultidelegationPage.ts (hunk 6 above)
Comment (oldGreg5) [reply packages/e2e-tests/src/elements/staking/MultidelegationPage.ts:66]: Yeah, normally I would prefer having less code for that simple thing(just to options) but we're gonna have there 3rd tab eventually for sure so why not, thanks.

File: packages/e2e-tests/src/steps/multidelegationSteps.ts (hunk 4 above)
Comment (oldGreg5) [reply packages/e2e-tests/src/steps/multidelegationSteps.ts:14]: It's for the visibility on a report

**Review by wklos-iohk — APPROVED** [review]

**Review by ljagiela — APPROVED** [review]

**Review by przemyslaw-wlodek — APPROVED** [review]

**Pull request merged**